test/test_source.vim

```diff
@@ -0,0 +1,31 @@
+let s:vimlparser = vimlparser#import()
+
+function! s:run()
+  let src = [
+  \  'function! s:load_on()',
+  \  '  imap <expr> <Tab> neosnippet#expandable_or_jumpable() ?',
+  \  '\ "\<Plug>(neosnippet_expand_or_jump)" : "\<Tab>"',
+  \  'endfunction',
+  \]
+  let r = s:vimlparser.StringReader.new(src)
+  let p = s:vimlparser.VimLParser.new(0)
+  let c = s:vimlparser.Compiler.new()
+  let toplevel = p.parse(r)
+  let func = toplevel.body[0]
+  let body = s:extract_body(func, src)
+  call writefile(split(body, "\n"), 'test/test_source.got')
+  qall!
+endfunction
+
+function! s:extract_body(func, src)
+  let pos = a:func.pos
+
+  let endpos = a:func.endfunction.pos
+  let endfunc = a:func.endfunction.ea
+  let cmdlen = endfunc.argpos.i - endfunc.cmdpos.i
```

Comment:
Using argpos to calculate the end of position is workaround so I think we should not use it in test officially. 
Can you change the test case or add comments?

P.S. We need to to add "endpos" field which contains the end of the position of node instead of this hack.

Reply:
agreed, but seems bits difficual to do in this issue. And we don't have time until tyru's advent calendar.